includes/class-webmention-receiver.php

```diff
@@ -382,25 +407,57 @@ public static function check_dupes( $commentdata ) {
 			return $commentdata;
 		}
 		$fragment = wp_parse_url( $commentdata['target'], PHP_URL_FRAGMENT );
+		if ( ! empty( $fragment ) ) {
+			// Check for the newer meta value before checking for the more traditional location
+			$args = array(
+				'post_id' => $commentdata['comment_post_ID'],
+				'meta_query' => array(
+					array(
+						'key' => 'webmention_source_url',
+						'value' => $commentdata['comment_author_url'],
+						'compare' => '=',
+					),
+					array(
+						'key' => 'webmention_target_fragment',
+						'value' => $fragment,
+						'compare' => '=',
+					),
+				),
+			);
+		} else {
+			$args = array(
+				'post_id' => $commentdata['comment_post_ID'],
+				'meta_key' => 'webmention_source_url',
+				'meta_value' => $commentdata['comment_author_url'],
+			);
+		}
+
+		$comments = get_comments( $args );
+		// check result
+		if ( ! empty( $comments ) ) {
+			$comment = $comments[0];
+			$commentdata['comment_ID'] = $comment->comment_ID;
+			$commentdata['comment_approved'] = $comment->comment_approved;
 
+			return $commentdata;
+		}
+
+		// Check in comment_author_url if the newer location is empty
 		$args = array(
 			'post_id' => $commentdata['comment_post_ID'],
-			'author_url' => esc_url_raw( $commentdata['comment_author_url'] ),
+			'author_url' => $commentdata['comment_author_url'],
```

Comment:
If you are referring to esc_url_raw...we actually call it when we set comment_author_url.